cmd/ci-operator-prowgen/main.go

```diff
@@ -27,6 +27,29 @@ const (
 	sentryDsnSecretName = "sentry-dsn"
 	sentryDsnMountPath  = "/etc/sentry-dsn"
 	sentryDsnSecretPath = "/etc/sentry-dsn/ci-operator"
+
+	openshiftInstallerRandomCmd = `set -eux
+target=$(awk < /usr/local/e2e-targets \
```

Comment:
Ugh, awk. [If only we had Python available](https://docs.python.org/3/library/random.html#random.choices) :(

I don't like the fact that we have duplicate lists of platforms (here and in the `/usr/local/e2e-targets` weights file). I'm thinking we could make `/usr/local/e2e-targets` a sourceable bash file, covering L32-L41 of this file. Then we would just source it here and it would set `template ` and `CLUSTER_TYPE` accordingly.

Reply:
A couple of points (in decreasing order of importance):
- What's wrong with awk? =)
- This is a quick-and-dirty (actually, the quickest-and-dirtiest) implementation: the goal of the task is to get the jobs in place as soon as possible, then refine. The job is generated by `prowgen`, so changing the implementation is trivial.
- I treated the format of `e2e-targets` as definitive. If/when this becomes Go code, maybe YAML would make more sense. We can bikeshed the exact format, but I went for "as-simple-as-possible plain text". I think we specially don't want to tie it to a specific implementation, as that is very likely to change.
- Isn't it available? I assumed it were, but even then I still think `awk` is unbeatable for ad-hoc text processing. The `python` version (with `random.choices`, an equivalent implementation would be even longer):

```python
import random, sys
ws, ns = zip(*((float(w), n) for (w, n) in map(str.split, sys.stdin)))
print(random.choices(ns, weights=ws)[0])
```

*sorry for the little essay, I wanted to answer other questions that were likely to come up*